components/agents/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java

```diff
@@ -707,19 +720,28 @@ protected String marshall(XMLObject xmlObject) throws SSOAgentException {
      * @return
      * @throws Exception
      */
-    protected Assertion getDecryptedAssertion(EncryptedAssertion encryptedAssertion) throws Exception {
-
-        KeyInfoCredentialResolver keyResolver = new StaticKeyInfoCredentialResolver(
-                new X509CredentialImpl(ssoAgentConfig.getSAML2().getSSOAgentX509Credential()));
-        EncryptedKey key = encryptedAssertion.getEncryptedData().
-                getKeyInfo().getEncryptedKeys().get(0);
-        Decrypter decrypter = new Decrypter(null, keyResolver, null);
-        SecretKey dkey = (SecretKey) decrypter.decryptKey(key, encryptedAssertion.getEncryptedData().
-                getEncryptionMethod().getAlgorithm());
-        Credential shared = SecurityHelper.getSimpleCredential(dkey);
-        decrypter = new Decrypter(new StaticKeyInfoCredentialResolver(shared), null, null);
-        decrypter.setRootInNewDocument(true);
-        return decrypter.decrypt(encryptedAssertion);
+    protected Assertion getDecryptedAssertion(EncryptedAssertion encryptedAssertion) throws SSOAgentException {
+
+        try {
+            KeyInfoCredentialResolver keyResolver = new StaticKeyInfoCredentialResolver(
+                    new X509CredentialImpl(ssoAgentConfig.getSAML2().getSSOAgentX509Credential()));
+
+            EncryptedKey key = encryptedAssertion.getEncryptedData().
+                    getKeyInfo().getEncryptedKeys().get(0);
+            Decrypter decrypter = new Decrypter(null, keyResolver, null);
+            SecretKey dkey = (SecretKey) decrypter.decryptKey(key, encryptedAssertion.getEncryptedData().
+                    getEncryptionMethod().getAlgorithm());
+            Credential shared = SecurityHelper.getSimpleCredential(dkey);
+            decrypter = new Decrypter(new StaticKeyInfoCredentialResolver(shared), null, null);
+            decrypter.setRootInNewDocument(true);
+            return decrypter.decrypt(encryptedAssertion);
+        } catch (Exception e){
```

Comment:
no need to add a debug log here, throw the exception with message 

ex. new SSOException("message", e);